src/sd/sd_client.cpp

```
@@ -193,13 +192,25 @@ class SdClientImpl : public transport::ITransportListener {
+        {
+            platform::ScopedLock const lock(eventgroup_subscriptions_mutex_);
+            const uint64_t key = (static_cast<uint64_t>(service_id) << 32U) |
+                                 (static_cast<uint64_t>(instance_id) << 16U) |
+                                 eventgroup_id;
+            EventGroupSubscription sub;
+            sub.service_id = service_id;
+            sub.instance_id = instance_id;
+            sub.eventgroup_id = eventgroup_id;
+            sub.major_version = 0x01;
+            eventgroup_subscriptions_[key] = sub;
+        }
```

Cppcheck (2.20.0)

[style] The function 'get_configuration_string' is never used.

(unusedFunction)